lib/ReactGridLayout.jsx

```diff
@@ -414,10 +418,58 @@ export default class ReactGridLayout extends React.Component<Props, State> {
       }
 
       if (!hasCollisions) {
-        // Set new width and height.
-        l.w = w;
-        l.h = h;
-      }
+        if (
+          [
+            "react-resizable-handle-sw",
+            "react-resizable-handle-w",
+            "react-resizable-handle-nw",
+            "react-resizable-handle-n",
+            "react-resizable-handle-ne"
+          ].indexOf(node.classList.item(1)) !== -1
+        ) {
+          let x = l.x;
+          let y = l.y;
+          if (
+            [
+              "react-resizable-handle-sw",
+              "react-resizable-handle-nw",
+              "react-resizable-handle-w"
+            ].indexOf(node.classList.item(1)) !== -1
+          ) {
+            x = l.x + (l.w - w);
+            x = x < 0 ? 0 : x;
+          }
+
+          if (
+            [
+              "react-resizable-handle-ne",
+              "react-resizable-handle-n",
+              "react-resizable-handle-nw"
+            ].indexOf(node.classList.item(1)) !== -1
+          ) {
+            y = l.y + (l.h - h);
+            y = y < 0 ? 0 : y;
+          }
+
+          l.w = w;
+          l.h = h;
+          // Move the element to the new position.
+          const isUserAction = true;
+          layout = moveElement(
```

Comment:
This should be done outside of `withLayoutItem`. Can it be represented as a drag instead?